### ISSUE TITLE
Validation Loss curve is different from vanilla nnUNet

### ISSUE THREAD

**Issue by steve-zeyu-zhang** [yhuang1997/3D-CPS#3]
Hi team,

I found something something confused.

Compared with the vanilla nnUNet, the validation Loss curve seems like does not align with the training loss curve (it doesn't decrease that much as training loss curve.) See below

3D-CPS:
![progress_3D-cps](https://github.com/yhuang1997/3D-CPS/assets/71418317/cb7d1890-598a-4b15-bfd4-401c07a36270)

vanilla nnUNet:

![unet](https://github.com/yhuang1997/3D-CPS/assets/71418317/f34c0222-5324-4526-92e4-84574b282a96)

This issue lets the performance cannot compared with vanilla nnUNet. I believe I must messed up something.

Is that because I'm using `FOLD = ['all']` for evaluation? If so, should I set a `splits_final.pkl` and let the `fold = 1` be the some labeled training data and use `fold = 1` to evaluate?

**Comment by yhuang1997** [comment]
The difference between training and online validation in the 3D-CPS(Fig. 1) is caused by different loss functions. We add CPS loss generated from unlabeled data on the original loss during the training stage (blue line) and keep using the original loss for online validation (red line). If you use `fold='all'` to train nnU-Net on your dataset, comparing loss on the online validation is meaningless. I advise you to run `inference` and record metrics to compare these two methods.

**Comment by steve-zeyu-zhang** [comment]
Hi team,

Thanks for the advice, I'm now doing a testing to try to use only a very small number of unlabeled and try to compare with the vanilla nnUNet to see if I can get my performance back on the right track lol.

Steve
